src/main/java/webserver/RequestHandler.java

```diff
@@ -1,46 +1,99 @@
 package webserver;
 
+import controller.UserController;
+import http.HttpBody;
+import http.HttpHeader;
+import http.HttpUtils;
+import http.request.HttpRequest;
+import http.request.HttpRequestLine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
 import java.io.DataOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.InputStreamReader;
 import java.io.OutputStream;
 import java.net.Socket;
-
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
 
 public class RequestHandler implements Runnable {
-    private static final Logger logger = LoggerFactory.getLogger(RequestHandler.class);
-
-    private Socket connection;
+    private static final Logger LOGGER = LoggerFactory.getLogger(RequestHandler.class);
+    private final UserController userController = new UserController();
+    private final Socket connection;
 
     public RequestHandler(Socket connectionSocket) {
         this.connection = connectionSocket;
     }
 
     public void run() {
-        logger.debug("New Client Connect! Connected IP : {}, Port : {}", connection.getInetAddress(),
+        LOGGER.debug("New Client Connect! Connected IP : {}, Port : {}", connection.getInetAddress(),
                 connection.getPort());
 
         try (InputStream in = connection.getInputStream(); OutputStream out = connection.getOutputStream()) {
-            // TODO 사용자 요청에 대한 처리는 이 곳에 구현하면 된다.
+            HttpRequest httpRequest = new HttpRequest();
+            BufferedReader reader = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8)); // HTTP Request 읽기
+            String startLine = reader.readLine(); // start line
+            HttpRequestLine httpRequestLine = new HttpRequestLine(startLine);
+            httpRequest.setRequestLine(httpRequestLine);
+            LOGGER.debug("--- HTTP Request : {} ---", httpRequestLine);
+
+            String line;
+            List<String> headers = new ArrayList<>();
+            while (!(line = reader.readLine()).equals("")) { // headers
```

Comment:
```suggestion
            while (!(line = reader.readLine()).isEmpty()) { // headers
```
위와 같이 isEmpty를 사용하여 더 가독성있게 만들 수 있을 것 같습니다.!!